lib/settings.js

```diff
@@ -0,0 +1,156 @@
+import _ from 'lodash';
+import logger from './logger';
+
+
+const SETTINGS_CAPS = [
+  'locationServicesEnabled',
+  'locationServicesAuthorized',
+];
+const SAFARI_SETTINGS_CAPS = [
+  'safariAllowPopups',
+  'safariIgnoreFraudWarning',
+  'safariOpenLinksInBackground',
+];
+
+async function launchAndQuitSimulator (sim, safari) {
+  logger.debug('No simulator directories found.');
+  return await sim.launchAndQuit(safari);
+}
+
+function checkPreferences (settings, opts = {}) {
+  for (let setting of settings) {
+    if (_.has(opts, setting)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// pass in the simulator so that other systems that use the function can supply
+// whatever they have
+async function setLocale (sim) {
```

Comment:
maybe we do or don't want to do this now, but we could refactor this so that it took parameters like opts and whether something is a real device, so that this becomes a self-contained function which doesn't need to be added to a prototype in order to be useful.